plugins/gfproxyd/rest.go

```diff
@@ -0,0 +1,152 @@
+package gfproxyd
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pborman/uuid"
+)
+
+func gfproxydEnableHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//validate volume name
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if gfproxyd is already enabled
+	if volume.IsGfproxydEnabled(volinfo) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrGfproxydAlreadyEnabled.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Transaction which starts gfproxyd on all nodes.
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	txn.Ctx.Set("volname", volname)
+
+	// Enable gfproxyd
+	volinfo.Options[volume.VkeyFeaturesGfproxyd] = "on"
```

Comment:
After this step, `txn.Ctx.Set("volinfo"` is required for  "vol-option.UpdateVolinfo" transaction step

Reply:
You shouldn't be directly setting options in a volinfo, without trying to verify that it is supported on all peers. The volume set step functions could be used to do the same. I don't know if its possible right now, but a TODO here will help remind us later.

Reply:
For now this behavior is fine for internal options.